L1Trigger/DTTriggerPhase2/plugins/DTTrigPhase2Prod.cc

```diff
@@ -0,0 +1,719 @@
+    mpathanalyzer_ = std::unique_ptr<MuonPathAnalyzerInChamber>(new MuonPathAnalyzerInChamber(pset, consumesColl));
+  }
+
+  // Getting buffer option
+  activateBuffer_ = pset.getParameter<bool>("activateBuffer");
+  superCellhalfspacewidth_ = pset.getParameter<int>("superCellspacewidth") / 2;
+  superCelltimewidth_ = pset.getParameter<double>("superCelltimewidth");
+
+  mpathqualityenhancer_ = std::unique_ptr<MPQualityEnhancerFilter>(new MPQualityEnhancerFilter(pset));
+  mpathredundantfilter_ = std::unique_ptr<MPRedundantFilter>(new MPRedundantFilter(pset));
+  mpathassociator_ = std::unique_ptr<MuonPathAssociator>(new MuonPathAssociator(pset, consumesColl));
+  rpc_integrator_ = std::unique_ptr<RPCIntegrator>(new RPCIntegrator(pset, consumesColl));
+
+  dtGeomH = esConsumes<DTGeometry, MuonGeometryRecord, edm::Transition::BeginRun>();
+}
+
+DTTrigPhase2Prod::~DTTrigPhase2Prod() {
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTp2: calling destructor" << std::endl;
+}
+
+void DTTrigPhase2Prod::beginRun(edm::Run const& iRun, const edm::EventSetup& iEventSetup) {
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTTrigPhase2Prod::beginRun " << iRun.id().run();
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTTrigPhase2Prod::beginRun: getting DT geometry";
+
+  grouping_obj_->initialise(iEventSetup);          // Grouping object initialisation
+  mpathanalyzer_->initialise(iEventSetup);         // Analyzer object initialisation
+  mpathqualityenhancer_->initialise(iEventSetup);  // Filter object initialisation
+  mpathredundantfilter_->initialise(iEventSetup);  // Filter object initialisation
+  mpathassociator_->initialise(iEventSetup);       // Associator object initialisation
+
+  const MuonGeometryRecord& geom = iEventSetup.get<MuonGeometryRecord>();
+  dtGeo_ = &geom.get(dtGeomH);
+}
+
+void DTTrigPhase2Prod::produce(Event& iEvent, const EventSetup& iEventSetup) {
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTTrigPhase2Prod::produce";
+  edm::Handle<DTDigiCollection> dtdigis;
+  iEvent.getByToken(dtDigisToken_, dtdigis);
+
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "\t Getting the RPC RecHits" << std::endl;
+  edm::Handle<RPCRecHitCollection> rpcRecHits;
+  iEvent.getByToken(rpcRecHitsLabel_, rpcRecHits);
+
+  ////////////////////////////////
+  // GROUPING CODE:
+  ////////////////////////////////
+  DTDigiMap digiMap;
+  DTDigiCollection::DigiRangeIterator detUnitIt;
+  for (detUnitIt = dtdigis->begin(); detUnitIt != dtdigis->end(); ++detUnitIt) {
+    const DTLayerId& layId = (*detUnitIt).first;
+    const DTChamberId chambId = layId.superlayerId().chamberId();
+    const DTDigiCollection::Range& range = (*detUnitIt).second;
+    digiMap[chambId].put(range, layId);
+  }
+
+  // generate a list muon paths for each event!!!
+  if (debug_ && activateBuffer_)
+    LogDebug("DTTrigPhase2Prod")
+        << "DTTrigPhase2Prod::produce - Getting and grouping digis per chamber using a buffer and super cells.";
```

Comment:
no need to repeat the producer name in the message category and message body (especially since the MessageLogger already knows which module emitted the message)